pkg/mimir/mimir.go

```diff
@@ -158,6 +160,70 @@ func (c *Config) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 	c.MemberlistKV.RegisterFlags(f)
 	c.ActivityTracker.RegisterFlags(f)
 	c.QueryScheduler.RegisterFlags(f)
+
+	c.Common.RegisterFlags(f)
+}
+
+func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	// First unmarshal common into the specific locations.
+	common := configWithCustomCommonUnmarshaler{
+		Common: &commonConfigUnmarshaler{
+			Storage: &specificLocationsUnmarshaler{
+				"blocks_storage":       &c.BlocksStorage.Bucket.StorageBackendConfig,
+				"ruler_storage":        &c.RulerStorage.StorageBackendConfig,
+				"alertmanager_storage": &c.AlertmanagerStorage.StorageBackendConfig,
+			},
+		},
+	}
+	if err := unmarshal(&common); err != nil {
+		return fmt.Errorf("can't unmarshal common config: %w", err)
+	}
+
+	// Then unmarshal config in a standard way.
```

Comment:
[nit] I would extend the comment to clarify that if the specific locations config options are set, this will override the common settings that were previously set, so that "YAML specific" takes precedence over "YAML common".

Reply:
Clarified in https://github.com/grafana/mimir/pull/2330/commits/e1a9253e4aba10141b8a8c1522cf190980fa5c2f